File: conversions/django_models/location.py
from django.db import models
from django.core.validators import MaxValueValidator
from .abstract_element_model import AbstractElementModel

class Location(AbstractElementModel):

    # Locality
    scene = models.TextField(blank=True, null=True)
    activity = models.TextField(blank=True, null=True)
    founding_date = models.PositiveIntegerField(blank=True, null=True)
    population_size = models.PositiveIntegerField(blank=True, null=True)
    parent_location = models.ForeignKey("Location", on_delete=models.SET_NULL, blank=True, null=True, related_name="location_parent_location")
    populations = models.ManyToManyField("Collective", blank=True, related_name="location_populations")

    # Culture
    traditions = models.TextField(blank=True, null=True)
    celebrations = models.TextField(blank=True, null=True)
    primary_cult = models.ForeignKey("Construct", on_delete=models.SET_NULL, blank=True, null=True, related_name="location_primary_cult")
    secondary_cults = models.ManyToManyField("Construct", blank=True, related_name="location_secondary_cults")
    delicacies = models.ManyToManyField("Species", blank=True, related_name="location_delicacies")

    # World
    coordinates = models.TextField(blank=True, null=True)
    founders = models.ManyToManyField("Character", blank=True, related_name="location_founders")

    # Construction
    logistics = models.TextField(blank=True, null=True)
    architecture = models.TextField(blank=True, null=True)
    construction_rate = models.PositiveIntegerField(validators=[MaxValueValidator(100)], blank=True, null=True)
    buildings = models.ManyToManyField("Location", blank=True, related_name="location_buildings")
    building_expertise = models.ManyToManyField("Construct", blank=True, related_name="location_building_expertise")

    # Production
    extraction = models.TextField(blank=True, null=True)
    industry = models.TextField(blank=True, null=True)
    extraction_output = models.PositiveIntegerField(blank=True, null=True)
    industry_output = models.PositiveIntegerField(blank=True, null=True)
    primary_resource = models.ForeignKey("Construct", on_delete=models.SET_NULL, blank=True, null=True, related_name="location_primary_resource")
    primary_industry = models.ForeignKey("Construct", on_delete=models.SET_NULL, blank=True, null=True, related_name="location_primary_industry")
    secondary_resources = models.ManyToManyField("Construct", blank=True, related_name="location_secondary_resources")
    secondary_industries = models.ManyToManyField("Construct", blank=True, related_name="location_secondary_industries")

    # Commerce
    trade = models.TextField(blank=True, null=True)
    infrastructure = models.TextField(blank=True, null=True)
    currency = models.TextField(blank=True, null=True)
    primary_extraction_market = models.ForeignKey("Location", on_delete=models.SET_NULL, blank=True, null=True, related_name="location_primary_extraction_market")
    primary_industry_market = models.ForeignKey("Location", on_delete=models.SET_NULL, blank=True, null=True, related_name="location_primary_industry_market")
    secondary_extraction_markets = models.ManyToManyField("Location", blank=True, related_name="location_secondary_extraction_markets")
    secondary_industry_markets = models.ManyToManyField("Location", blank=True, related_name="location_secondary_industry_markets")

    # Localpolitics
    government = models.TextField(blank=True, null=True)
    opposition = models.TextField(blank=True, null=True)
    governing_title = models.ForeignKey("Title", on_delete=models.SET_NULL, blank=True, null=True, related_name="location_governing_title")
    primary_faction = models.ForeignKey("Institution", on_delete=models.SET_NULL, blank=True, null=True, related_name="location_primary_faction")
    secondary_factions = models.ManyToManyField("Institution", blank=True, related_name="location_secondary_factions")

    # Regionalpolitics
    territorial_policies = models.TextField(blank=True, null=True)
    territory = models.ForeignKey("Territory", on_delete=models.SET_NULL, blank=True, null=True, related_name="location_territory")
    rival = models.ForeignKey("Location", on_delete=models.SET_NULL, blank=True, null=True, related_name="location_rival")
    friend = models.ForeignKey("Location", on_delete=models.SET_NULL, blank=True, null=True, related_name="location_friend")
    soft_influence_on = models.ManyToManyField("Location", blank=True, related_name="location_soft_influence_on")
    hard_influence_on = models.ManyToManyField("Location", blank=True, related_name="location_hard_influence_on")

    # Strategics
    defensibility = models.TextField(blank=True, null=True)
    height = models.PositiveIntegerField(blank=True, null=True)
    primary_fighter = models.ForeignKey("Institution", on_delete=models.SET_NULL, blank=True, null=True, related_name="location_primary_fighter")
    secondary_fighters = models.ManyToManyField("Institution", blank=True, related_name="location_secondary_fighters")
    defenses = models.ManyToManyField("Location", blank=True, related_name="location_defenses")
    fortifications = models.ManyToManyField("Object", blank=True, related_name="location_fortifications")

    def __str__(self):
        return self.name

File: conversions/csharp_unity_json_properties/Location.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Utils;

[System.Serializable]
public class Location : Element
{
    [JsonProperty("scene"), TextAttribute("")]
    public string scene;
    [JsonProperty("activity"), TextAttribute("")]
    public string activity;
    [JsonProperty("founding_date"), Integer(0)]
    public int foundingDate;
    [JsonProperty("population_size"), Integer(0)]
    public int populationSize;
    [JsonProperty("parent_location"), ReferenceAttribute(typeof(Location))]
    public string parentLocation;
    [JsonProperty("populations"), ReferenceAttribute(typeof(Collective), true)]
    public string populations;
    [JsonProperty("traditions"), TextAttribute("")]
    public string traditions;
    [JsonProperty("celebrations"), TextAttribute("")]
    public string celebrations;
    [JsonProperty("primary_cult"), ReferenceAttribute(typeof(Construct))]
    public string primaryCult;
    [JsonProperty("secondary_cults"), ReferenceAttribute(typeof(Construct), true)]
    public string secondaryCults;
    [JsonProperty("delicacies"), ReferenceAttribute(typeof(Species), true)]
    public string delicacies;
    [JsonProperty("coordinates"), TextAttribute("")]
    public string coordinates;
    [JsonProperty("founders"), ReferenceAttribute(typeof(Character), true)]
    public string founders;
    [JsonProperty("logistics"), TextAttribute("")]
    public string logistics;
    [JsonProperty("architecture"), TextAttribute("")]
    public string architecture;
    [JsonProperty("construction_rate"), Integer(100)]
    public int constructionRate;
    [JsonProperty("buildings"), ReferenceAttribute(typeof(Location), true)]
    public string buildings;
    [JsonProperty("building_expertise"), ReferenceAttribute(typeof(Construct), true)]
    public string buildingExpertise;
    [JsonProperty("extraction"), TextAttribute("")]
    public string extraction;
    [JsonProperty("industry"), TextAttribute("")]
    public string industry;
    [JsonProperty("extraction_output"), Integer(0)]
    public int extractionOutput;
    [JsonProperty("industry_output"), Integer(0)]
    public int industryOutput;
    [JsonProperty("primary_resource"), ReferenceAttribute(typeof(Construct))]
    public string primaryResource;
    [JsonProperty("primary_industry"), ReferenceAttribute(typeof(Construct))]
    public string primaryIndustry;
    [JsonProperty("secondary_resources"), ReferenceAttribute(typeof(Construct), true)]
    public string secondaryResources;
    [JsonProperty("secondary_industries"), ReferenceAttribute(typeof(Construct), true)]
    public string secondaryIndustries;
    [JsonProperty("trade"), TextAttribute("")]
    public string trade;
    [JsonProperty("infrastructure"), TextAttribute("")]
    public string infrastructure;
    [JsonProperty("currency"), TextAttribute("")]
    public string currency;
    [JsonProperty("primary_extraction_market"), ReferenceAttribute(typeof(Location))]
    public string primaryExtractionMarket;
    [JsonProperty("primary_industry_market"), ReferenceAttribute(typeof(Location))]
    public string primaryIndustryMarket;
    [JsonProperty("secondary_extraction_markets"), ReferenceAttribute(typeof(Location), true)]
    public string secondaryExtractionMarkets;
    [JsonProperty("secondary_industry_markets"), ReferenceAttribute(typeof(Location), true)]
    public string secondaryIndustryMarkets;
    [JsonProperty("government"), TextAttribute("")]
    public string government;
    [JsonProperty("opposition"), TextAttribute("")]
    public string opposition;
    [JsonProperty("governing_title"), ReferenceAttribute(typeof(Title))]
    public string governingTitle;
    [JsonProperty("primary_faction"), ReferenceAttribute(typeof(Institution))]
    public string primaryFaction;
    [JsonProperty("secondary_factions"), ReferenceAttribute(typeof(Institution), true)]
    public string secondaryFactions;
    [JsonProperty("territorial_policies"), TextAttribute("")]
    public string territorialPolicies;
    [JsonProperty("territory"), ReferenceAttribute(typeof(Territory))]
    public string territory;
    [JsonProperty("rival"), ReferenceAttribute(typeof(Location))]
    public string rival;
    [JsonProperty("friend"), ReferenceAttribute(typeof(Location))]
    public string friend;
    [JsonProperty("soft_influence_on"), ReferenceAttribute(typeof(Location), true)]
    public string softInfluenceOn;
    [JsonProperty("hard_influence_on"), ReferenceAttribute(typeof(Location), true)]
    public string hardInfluenceOn;
    [JsonProperty("defensibility"), TextAttribute("")]
    public string defensibility;
    [JsonProperty("height"), Integer(0)]
    public int height;
    [JsonProperty("primary_fighter"), ReferenceAttribute(typeof(Institution))]
    public string primaryFighter;
    [JsonProperty("secondary_fighters"), ReferenceAttribute(typeof(Institution), true)]
    public string secondaryFighters;
    [JsonProperty("defenses"), ReferenceAttribute(typeof(Location), true)]
    public string defenses;
    [JsonProperty("fortifications"), ReferenceAttribute(typeof(Object), true)]
    public string fortifications;
}
